src/runtime/TemporalObject.cpp

```diff
@@ -889,22 +944,36 @@ std::map<TemporalObject::DateTimeUnits, int> TemporalPlainTimeObject::toTemporal
     std::map<TemporalObject::DateTimeUnits, int> result = { { TemporalObject::HOUR_UNIT, 0 }, { TemporalObject::MINUTE_UNIT, 0 }, { TemporalObject::SECOND_UNIT, 0 }, { TemporalObject::MILLISECOND_UNIT, 0 }, { TemporalObject::MICROSECOND_UNIT, 0 }, { TemporalObject::NANOSECOND_UNIT, 0 } };
     TemporalObject::DateTimeUnits temporalTimeLikeProp[6] = { TemporalObject::HOUR_UNIT, TemporalObject::MICROSECOND_UNIT, TemporalObject::MILLISECOND_UNIT, TemporalObject::MINUTE_UNIT, TemporalObject::NANOSECOND_UNIT, TemporalObject::SECOND_UNIT };
 
-    bool any = false;
+    ValueVector values = { Value(state.context()->staticStrings().lazyHour().string()),
+                           Value(state.context()->staticStrings().lazymicrosecond().string()),
+                           Value(state.context()->staticStrings().lazymillisecond().string()),
+                           Value(state.context()->staticStrings().lazyMinute().string()),
+                           Value(state.context()->staticStrings().lazynanosecond().string()),
+                           Value(state.context()->staticStrings().lazySecond().string()) };
+    Value partial = TemporalCalendarObject::prepareTemporalFields(state, temporalTimeLike, values, {});
+
     for (auto i : temporalTimeLikeProp) {
-        Value value = temporalTimeLike.asObject()->get(state, ObjectPropertyName(state, new ASCIIString(dateTimeUnitStrings[i]))).value(state, temporalTimeLike);
-        any = !value.isUndefined();
+        ObjectHasPropertyResult objectHasPropertyResult = partial.asObject()->hasProperty(state, ObjectPropertyName(state, new ASCIIString(dateTimeUnitStrings[i])));
```

Comment:
About `dateTimeUnitStrings` string array
In fact, each element of this string array is already defined in static strings.
For example, we can easily get `year` string like below.
```c++
state.context()->staticStrings().lazyYear(); // get 'year' AtomicString
state.context()->staticStrings().lazyYear().string(); // get 'year' Stringobject
```

So, what about replacing this global array with already defined static strings?